packages/react-devtools-extensions/src/main.js

```diff
@@ -25,6 +25,25 @@ const LOCAL_STORAGE_SUPPORTS_PROFILING_KEY =
 
 const isChrome = getBrowserName() === 'Chrome';
 
+const cachedNetworkEvents = new Map();
+
+// Cache JavaScript resources as the page loads them.
+// This helps avoid unnecessary duplicate requests when hook names are parsed.
+// Responses with a Vary: 'Origin' migt not match future requests.
+// This lets us avoid a possible (expensive) cache miss.
+// For more info see: github.com/facebook/react/pull/22198
+chrome.devtools.network.onRequestFinished.addListener(
+  function onRequestFinished(event) {
+    const {method, url} = event.request;
+    if (method === 'GET' && url.indexOf('.js') > 0) {
```

Comment:
JavaScript files aren't guaranteed to contain `.js` as the web doesn't care about file extensions, only MIME types. A place I used to work at a long time ago (before JS modules existed) had a custom ASP.NET-based bundler and the URLs were like `/scripts.ashx/MainSite`, but that's not uncommon today too.

Since you're checking the MIME type below, it should be safe to just remove this extension check.

Reply:
Good point.